main.js

```diff
@@ -3706,6 +3720,14 @@ let Sites = {
 					originX /= spawns.length;
 					originY /= spawns.length;
 
+					//Ensure we have a spawn, otherwise return 25, 25
+					if(!originX || !originY)
+					{
+						console.log(`<font color=\"#FF0000\">[Invasion]</font> Could not detect any spawns in room ${rmColony}`);
+						originX = 25;
+						originY = 25;
```

Comment:
Probably best to include towers in calculating spawn origins, rather than just assuming base origin is 25,25 ... since origin should be useful.

Reply:
This is just a band-aid to prevent the script from halting if theres an enemy in a room with no spawns built.